sdk/attestation/azure-security-attestation/azure/security/attestation/_models.py

```diff
@@ -69,242 +79,244 @@ class PolicyCertificatesModificationResult(object):
     """
 
     def __init__(self, certificate_thumbprint, certificate_resolution):
-        #type:(str, CertificateModification)->None
+        # type: (str, CertificateModification, AttestationToken) -> None
         self.certificate_thumbprint = certificate_thumbprint
         self.certificate_resolution = certificate_resolution
 
     @classmethod
     def _from_generated(cls, generated):
+        # type: (GeneratedPolicyCertificatesModificationResult) -> AttestationPolicyCertificateResult
         if not generated:
-            return cls
+            return None
         return cls(generated.certificate_thumbprint, generated.certificate_resolution)
 
-class PolicyResult(object):
-    """ PolicyResult represents the result of a :meth:`azure.security.attestation.AttestationAdministrationClient.set_policy` 
-    or :meth:`azure.security.attestation.AttestationAdministrationClient.reset_policy`  API call.
 
-    The `PolicyResult` class is returned as the body of an attestation token from
+class AttestationPolicyResult(object):
+    """AttestationPolicyResult represents the result of a
+    :meth:`azure.security.attestation.AttestationAdministrationClient.set_policy`
+    or :meth:`azure.security.attestation.AttestationAdministrationClient.reset_policy`
+    API call.
+
+    The `AttestationPolicyResult` class is returned as the body of an attestation token from
     the attestation service. It can be used to ensure that the attestation service
     received the policy object sent from the client without alteration.
-    
-    :param policy_resolution: The result of the policy set or
-        reset call.
+
+    :param policy_resolution: The result of the policy set or reset call.
     :type policy_resolution: azure.security.attestation.PolicyModification
     :param policy_signer: If the call to `set_policy` or `reset_policy`
-        had a :class:`AttestationSigningKey` parameter, this will be the certificate
+        had a `signing_certificate` parameter, this will be the certificate
         which was specified in this parameter.
     :type policy_signer: azure.security.attestation.AttestationSigner
```

Comment:
I'm a bit confused here -- it seems like a JSONWebKey (the generated type) is expected here based on the code and type hint, but this asks for an AttestationSigner. The JSONWebKey model doesn't seem to be public, so it doesn't make much sense to ask for it here if the user is constructing this type themselves. Is this type used, but not constructed by a user, hence the oddity? Or am I missing some context?

Reply:
The `policy_signer` attribute is an `AttestationSigner`.

From what I've seen, the :param: element is used to document attributes on an object AND to document the parameter type. In this case, it's documenting the attribute.

The value returned from the service is a `JSONWebKey`, the constructor converts from `JSONWebKey` to `AttestationSigner.

Reply:
I'm just fixing this to do the work in from_generated. THat cleans up the issue.